nengo_viz/server.py

```diff
@@ -45,9 +45,18 @@ def swi_shutdown(self, *path):
     def ws_viz_component(self, client, uid):
         """Handles ws://host:port/viz_component with a websocket"""
         # figure out what component is being connected to
+
+        viz_sim = Server.viz_sim
+
+        component = viz_sim.uids[uid]
         try:
             while True:
-                component = self.viz_sim.uids[uid]
+                if viz_sim.finished:
+                    component.finish()
```

Comment:
Sorry I'm late to the game, but it seems strange to have this here since it's in the `finally` block.